### ISSUE TITLE
Reply with service-unavailable instead of feature-not-implemented to unkonwn IQs

### ISSUE THREAD

**Issue by horazont** [horazont/aioxmpp#75]
It appears that this is in fact the default behaviour we should show. We need to investigate. See xsf@ MUC logs:

```
(09:37:21) Ge0rG: Badly written XEPs for 100: "If the pinged entity does not support the ping namespace, it MUST return a <service-unavailable/> error: [example skipped] The other error conditions defined in RFC 6120 could also be returned if appropriate."
(09:40:09) jonasw: I thought the default response for non-supported namespaces is <feature-not-implemented/>?
(09:40:42) vanitasvitae: A general question: I often read "If the Entity does not support XYZ, it MUST return a <service-unavailable/> error." How is this done in real world implementations? Do the developers return this error whenever they encounter a stanza their client does not understand? Or do the devs have to go through every XEP and add the error for those XEPS they dont implement?
(09:40:46) Ge0rG: It's also nice how the XEP tells implementations that don't support the XEP what to do.
(09:41:08) Ge0rG: vanitasvitae: it's just bad wording.
(09:41:22) vanitasvitae: GeOrG: Thats exactly what I mean
(09:41:28) Ge0rG: vanitasvitae: if you don't implement an XEP, you can't be fored by the XEP to do anything.
(09:41:34) jonasw: In aioxmpp, I simply return <feature-not-implemented/>, is that incorrect?
(09:42:21) Ge0rG: jonasw: you MUST return service-unavailable, and you may also return something else. So you need to send an error response with two error conditions :P
(09:42:38) jonasw: let’s do that. no seriously, isn’t there wording in the RFCs what to do?
(09:58:01) Flow: vanitasvitae: That is the default IQ error reply if the entity does not support the IQ
(09:58:18) vanitasvitae: Ah okay.
(09:58:35) Flow: see https://github.com/igniterealtime/Smack/blob/c8e7c4804b79435ba36bc26c59e972d0fdc7a9fa/smack-core/src/main/java/org/jivesoftware/smack/AbstractXMPPConnection.java#L1097
(09:59:05) Flow: that "respond with service-unavailable" requirement is usually only for IQs
```

**Comment by horazont** [comment]
We still need to evaluate if there’s any authoritative source on this. From a quick glimpse at RFC 6120, I could not find any normative text regarding to that.

It appears however that some XEPs suggest to answer with ``<service-unavailable/>`` if the XEP is not supported (this is obviously not normative, because if the XEP is not supported, it has no power over the implementation).

**Comment by horazont** [comment]
https://tools.ietf.org/html/rfc6120#section-8.4
>        o  If an intended recipient receives an IQ stanza of type "get" or
>           "set" containing a child element qualified by a namespace it does
>           not understand, then the entity MUST return an IQ stanza of type
>           "error" with an error condition of <service-unavailable/>.

**Comment by horazont** [comment]
Fixed by 61eae3b4.
